src/openzeppelin/account/interface.cairo

```diff
@@ -12,10 +13,15 @@ struct Call {
     calldata: Array<felt252>
 }
 
-trait IAccount {
+trait IBaseAccount {
     fn __execute__(calls: Array<Call>) -> Array<Span<felt252>>;
     fn __validate__(calls: Array<Call>) -> felt252;
+}
+
+trait IDeclarer {
     fn __validate_declare__(class_hash: felt252) -> felt252;
-    fn is_valid_signature(message: felt252, signature: Array<felt252>) -> u32;
-    fn supports_interface(interface_id: u32) -> bool;
+}
```

Comment:
Removing `__validate_declare__` as a necessity for the base account makes sense to me. I'm not sure about giving it its own standalone interface though. It's not meant to be called by other contracts (similar to `__validate_deploy__`), right?

Reply:
That's right. Good call

Reply:
this is the reason why i'm not convinced about using traits for signaling ABIs. i do think we need this trait, not for other contracts to call (ABI) but for contracts to implement a shared behavior

the `#[abi]` directive is the one we use when we want something to be called from other contracts since it's the one that generates the associated dispatchers

Reply:
I added the trait thinking of that (extensibility for other modules), but I agreed with Andrew mostly because I think that having a lot of interfaces with just one method could be counterproductive, but I'm not sure where is the line, because ERC165 and ERC1271 are examples of interfaces with just one method that makes sense to implement, that's why I added IDeclarer as well.

Reply:
>  i do think we need this trait, not for other contracts to call (ABI) but for contracts to implement a shared behavior

Right, that's my bad. If we take that further though, shouldn't we also include IDeployer? And to Eric's point, I can also see the potential of it being counterproductive...or maybe it's just growing pains with getting used to the concept, idk

Reply:
I didn't add IDeployer because that interface is not meant to be used outside of just once in deployment (I wouldn't say it is part of the public API, but just a one-time protocol feature for deployment). IDeclarer on the other hand can be used many times on-chain to declare different contracts